File: tree/leetcode_Maximum_Depth_of_Binary_Tree.java
/*Maximum_Depth_of_Binary_Tree

Given a binary tree, find its maximum depth.
The maximum depth is the number of nodes along the longest path from the root
node down to the farthest leaf node.
*/

public class leetcode_Maximum_Depth_of_Binary_Tree {

    public static void main(String[] args) {

    }

    public static int maxDepth(TreeNode root) {
        if (root == null) {
			return 0;
		} else {
			return Math.max(maxDepth(root.right), maxDepth(root.left)) + 1;
		}
    }
}

File: tree/leetcode_Maximum_Depth_of_Binary_Tree.cpp
/**
 * Copyright 2016
 * author: madarfacar
 *
 * leetcode maximum depth of binary tree
 */

#include <stdio.h>
#include <algorithm>

/**
 * Definition for a binary tree node.
 */
struct TreeNode {
  int val;
  TreeNode *left;
  TreeNode *right;
  TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};


class Solution {
 public:
  int maxDepth(TreeNode* root) {
    if (root == NULL) {
      return 0;
    } else {
      int left_depth = maxDepth(root->left);
      int right_depth = maxDepth(root->right);
      return std::max(left_depth, right_depth) + 1;
    }
  }
};


int main() {
  Solution sol;

  TreeNode* root = new TreeNode(1);
  root->left = new TreeNode(1);

  printf("Depth: %d\n", sol.maxDepth(root));
}
